demuxer.go

```diff
@@ -23,21 +23,28 @@ var (
 // http://seidl.cs.vsb.cz/download/dvb/DVB_Poster.pdf
 // http://www.etsi.org/deliver/etsi_en/300400_300499/300468/01.13.01_40/en_300468v011301o.pdf
 type Demuxer struct {
-	ctx              context.Context
-	dataBuffer       []*DemuxerData
-	l                astikit.CompleteLogger
+	ctx        context.Context
+	dataBuffer []*DemuxerData
+	l          astikit.CompleteLogger
+
 	optPacketSize    int
 	optPacketsParser PacketsParser
-	packetBuffer     *packetBuffer
-	packetPool       *packetPool
-	programMap       *programMap
-	r                io.Reader
+	optPacketFilter  PacketFilter
+
+	packetBuffer *packetBuffer
+	packetPool   *packetPool
+	programMap   *programMap
+	r            io.Reader
 }
 
 // PacketsParser represents an object capable of parsing a set of packets containing a unique payload spanning over those packets
 // Use the skip returned argument to indicate whether the default process should still be executed on the set of packets
 type PacketsParser func(ps []*Packet) (ds []*DemuxerData, skip bool, err error)
 
+// PacketFilter executed right before pushing byte-payload to Packet-object. PacketHeader and PacketAdaptationField still be present
+// Use the skip to filter-out unwanted packets from pipeline. NextPacket() will return next unfiltered packet in query
+type PacketFilter func(p *Packet) (skip bool)
```

Comment:
Could you rename this to `PacketSkipper`?

Also could you change the above comment to:
```PacketSkipper represents an object capable of skipping a packet before parsing its payload. Its header and adaptation field is parsed and provided to the object.
Use this option if you need to filter out unwanted packets from your pipeline. NextPacket() will return the next unfiltered packet if any.```

Reply:
Done